OSDOCS-17165-graceful-shutdown CQA (#117010)
openshift/openshift-docs: modules/graceful-shutdown.adoc

[id="graceful-shutdown_{context}"]
= Shutting down the cluster

[role="_abstract"]

> Shut down the cluster by cordoning nodes, draining workloads, and stopping nodes in order. Graceful shutdown preserves cluster data so you can restart the cluster later.

Reviewer: >Shut down the cluster by cordoning nodes, draining workloads, and stopping nodes in order

I don't do these things, right? The graceful shutdown does, yeah?

Author: Created a new abstract. Hope it's better

[NOTE]
====
You can shut down a cluster until a year from the installation date and expect it to restart gracefully. After a year from the installation date, the cluster certificates expire. However, you might need to manually approve the pending certificate signing requests (CSRs) to recover kubelet certificates when the cluster restarts.
====

Take an etcd backup before shutting down the cluster so that you can restore the cluster if you encounter issues when restarting it.

> You might need the backup if any of the following occur:

Reviewer: ```suggestion
You might need to restore from the backup if any of the following conditions occur:
```

.Example output
[source,terminal]
----

> 2022-08-05T14:37:50Z

Reviewer: Maybe update this date?

```suggestion
2030-08-05T14:37:50Z
```

node/ci-ln-mgdnf4b-72292-n547t-worker-c-vcmtn cordoned
----

. Evacuate the pods by running the following command:
[source,terminal]
+
----
$ for node in $(oc get nodes -l node-role.kubernetes.io/worker -o jsonpath='{.items[*].metadata.name}'); do echo ${node} ; oc adm drain ${node} --delete-emptydir-data --ignore-daemonsets=true --timeout=15s --force ; done
----

. Shut down all of the nodes in the cluster by running the following command:
+
[source,terminal]
----
$ for node in $(oc get nodes -o jsonpath='{.items[*].metadata.name}'); do oc debug node/${node} -- chroot /host shutdown -h 1; done
----
+
[NOTE]
====
Ensure that the control plane node with the API VIP assigned is the last node processed in the loop. Otherwise, the shutdown command fails.
====
+

> Shutting down the nodes by using one of these methods allows pods to stop gracefully, which reduces the chance for data corruption.

Reviewer: Not sure this belongs here. Might be better in the intro section? (Without the _by using one of these methods_ as there is only one method shown?)

```suggestion
Shutting down the nodes by using this method allows pods to stop gracefully, which reduces the chance for data corruption.
```

Author: Moved this under the evacuate pods step and reworded it
